gobblin-service/src/test/java/org/apache/gobblin/service/GobblinServiceManagerTest.java

```diff
@@ -339,60 +342,83 @@ public void testRunOnceJob() throws Exception {
     AssertWithBackoff.create().maxSleepMs(200L).timeoutMs(2000L).backoffFactor(1)
         .assertTrue(input -> this.gobblinServiceManager.getFlowCatalog().getSpecs().size() == 0,
           "Waiting for job to get orchestrated...");
-    AssertWithBackoff.create().maxSleepMs(100L).timeoutMs(1000L).backoffFactor(1)
-          .assertTrue(input -> !this.gobblinServiceManager.getScheduler().getScheduledFlowSpecs().containsKey(TEST_URI.toString()),
-              "Waiting for job to get orchestrated...");
+    AssertWithBackoff.create().maxSleepMs(100L).timeoutMs(2000L).backoffFactor(1)
+          .assertTrue(input -> !this.gobblinServiceManager.getScheduler().getScheduledFlowSpecs().containsKey(uri.toString()),
```

Comment:
"not present" tests can mislead us.  e.g. wouldn't we find success here even if `uri.equals("what, me worry?")`?

should we first verify prior that the scheduler actually at one point had this URI?

Reply:
I tried to verify it's present immediately after but I think it's processed immediately. It seems too timing dependent to have two assert with backoff statements, the first to check it's there, then to check it disappears. I'm not sure they would work.